src/openms/source/ANALYSIS/MAPMATCHING/FeatureGroupingAlgorithm.cpp

```diff
@@ -43,6 +43,27 @@
 #include <OpenMS/CONCEPT/LogStream.h>
 #include <OpenMS/CONCEPT/Factory.h>
 
+#include <unordered_set>
+#include <regex>
+#include <map>
+
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/connected_components.hpp>
+
+struct VertexLabel
+{
+    VertexLabel() = default;
+    VertexLabel(OpenMS::String u, bool is_feat):uid(u), is_feature(is_feat) {}
+    OpenMS::String uid = "0";
```

Comment:
Also comment what it means when it is not a feature.

Reply:
Maybe copy parts of the description in annotateIonIdentityNetworks

Reply:
It's a String because the original "Group" meta value from the FeatureMap is already a String. And because the VertexLabel needs to work for both the feature and group vertex, it is a String here.

Reply:
Tried earlier to convert the String meta value to a size_t, but there were some issues. This should be possible without loss of information?

Reply:
@timosachsenberg should know. In theory yes, but not sure how complete the DataValue conversions are. I think I had problems before too.

Reply:
I think the question is if Meta/DataValue can handle this.

Reply:
ok if that is the question I agree. still would be possible to change the type in the vertex

Reply:
@axelwalter This basically means you cannot use the automatic conversions from DataValue. You need to use stoi or something like this.
I prefer uints/size_t in the vertex